react-components/src/architecture/base/README.md

```diff
@@ -0,0 +1,186 @@
+
+  - Public constants (exported)
+  - Instance fields.
+  - Instance properties
+  - Implementation of interface (if any)
+  - Overrides from the must basic base class (if any)
+  - :
+  - Overrides from the actually base class (if any)
+  - Instance methods (If many, break it up based on what they do)
+  - Static methods (if any)
+  - Local functions, types or classes used in this file only
+
+- **Virual methods:**: Since TypeScript doesn't allow the virtual keyword, all functions are virtual. This is a weekness for the application developer. I have tried to mark function as virtual by comments. You should not override other function than this. A virtual function normally brings in uncessesary complexibility and should general be used by care.
+
+  - Function with prefex `Core` are always virtual and protected. They can be overridden, but must call the same function for the base class. A function without the `Core` prefex is the one you should call from outside and is public. Experince shows that this pattern will make the code easier to maintain and extend in the future. For instance `remove/removeCore` and `initialize/initializeCore` in the `DomainObject`.
+  - Overridden methods should alway be marked by the override keyword. You will unfortunate not get any linting error when you forget this.
+  -
+
+- **Static methods and classes:** The linker don't like static method, epecially when the entire class have static method only. I wanted to use this more, but have used functions instead. I don't like this, because it doen't group similar function togeter. It will also be less readabble on the caller side, since you don't know what part of the code the function is coming from. In typescript we have for instance Math with act like a static class, but they use som magic to pretend it to be using interface.
+
+- **Reuse:** I try to reuse code whenever it is possible. Therefore you will find functions that are very small,
+  often one liners. This ensures the complexity of the code to be low and increase readability on the caller side. For instance:
+
+        export function clear<T>(array: T[]): void {
+            array.splice(0, array.length);
+        }
+
+- **Single responsibility:** I try to give each class one single responsibility. Therefore I use several base classes, each class adds some complexity.
+  Examples of this is `BaseView/ThreeView/GroupThreeView` which are all base classes. If they are merge to one single class, it will be too complex.
+
+- **Interpendence:** Try to keep must classes independed of other classes. For instance the tool should not know abount the views. Utility function and classes can used whenevery you like.
+
+- **Utility function and classes:** I have added som utility function/classes in `architecture/base/utilities`.
+  - Some utility functions may be a duplicate of some function in a package we use. Please tell me.
```

Comment:
`lodash` has a lot of simple util functions like this, we can probably go through them

Reply:
Good